calc/active_rules_calculator.go

```diff
@@ -184,6 +193,23 @@ func (arc *ActiveRulesCalculator) OnUpdate(update api.Update) (_ bool) {
 	return
 }
 
+func (arc *ActiveRulesCalculator) OnStatusUpdate(status api.SyncStatus) {
+	if status == api.InSync && !arc.datastoreInSync {
+		arc.datastoreInSync = true
+		if arc.missingProfiles.Len() > 0 {
+			// Log out any profiles that were missing during the resync.  We defer
+			// this until now because we may hear about profiles or endpoints first.
+			arc.missingProfiles.Iter(func(item interface{}) error {
+				log.WithField("profileID", item).Warning(
+					"End of resync: local endpoints refer to missing " +
+						"or invalid profile, profile's rules replaced " +
+						"with drop rules.")
+				return set.RemoveItem
+			})
+		}
+	}
```

Comment:
Isn't it possible for Felix to get out of sync again, after an initial in-sync?  If so, we should handle the possibility here of setting arc.datastoreInSync back to False.